src/databricks/labs/remorph/install.py

```diff
@@ -0,0 +1,191 @@
+import logging
+import os
+import webbrowser
+from datetime import timedelta
+from pathlib import Path
+
+from databricks.labs.blueprint.entrypoint import get_logger
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.installer import InstallState
+from databricks.labs.blueprint.tui import Prompts
+from databricks.labs.blueprint.wheels import ProductInfo
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+
+from databricks.labs.remorph.__about__ import __version__
+from databricks.labs.remorph.config import MorphConfig
+
+logger = logging.getLogger(__name__)
+
+PRODUCT_INFO = ProductInfo(__file__)
+
+
+class WorkspaceInstaller:
+    def __init__(self, prompts: Prompts, installation: Installation, ws: WorkspaceClient):
+        if "DATABRICKS_RUNTIME_VERSION" in os.environ:
+            msg = "WorkspaceInstaller is not supposed to be executed in Databricks Runtime"
+            raise SystemExit(msg)
+        self._ws = ws
+        self._installation = installation
+        self._prompts = prompts
+        self._catalog_setup = CatalogSetup(ws)
+
+    def run(self):
+        logger.info(f"Installing Remorph v{PRODUCT_INFO.version()}")
+        config = self.configure()
+        workspace_installation = WorkspaceInstallation(
+            config,
+            self._installation,
+            self._ws,
+            self._prompts,
+            verify_timeout=timedelta(minutes=2),
+        )
+        workspace_installation.run()
+
+    def configure(self) -> MorphConfig:
+        try:
+            return self._installation.load(MorphConfig)
+        except NotFound as err:
+            logger.debug(f"Cannot find previous installation: {err}")
+        logger.info("Please answer a couple of questions to configure Remorph")
+
+        source_prompt = self._prompts.choice("Select the source", ["snowflake", "tsql"])
+        source = source_prompt.lower()
+
+        skip_validation_str = self._prompts.choice_from_dict(
+            "Do you want to Skip Validation", {"Yes": "true", "No": "false"}
+        )
+        skip_validation = skip_validation_str == "true"  # convert to bool
+
+        catalog_name = self._prompts.question("Enter catalog_name")
+
+        if not self._catalog_setup.get(catalog_name):
+            allow_catalog_creation = self._prompts.question(
+                f"""Catalog {catalog_name} not found.\
+                    \nDo you want to create a new one?"""
+            )
+            if allow_catalog_creation:
+                self._catalog_setup.create(catalog_name)
+            else:
+                msg = "Catalog is needed to setup Remorph"
+                raise SystemExit(msg)
```

Comment:
refactor to a method and do a `@retried(on=[...`  for it

Reply:
Done